homeassistant/components/broadlink/sensor.py

```diff
@@ -144,31 +151,16 @@ def __init__(self, api, check_sensors, interval):
                 vol.Optional("noise"): vol.Any(0, 1, 2),
             }
         )
-        self.update = Throttle(interval)(self._update)
-        if not self._auth():
-            _LOGGER.warning("Failed to connect to device")
+        self.async_update = Throttle(interval)(self._async_fetch_data)
 
-    def _update(self, retry=3):
+    async def _async_fetch_data(self):
+        """Fetch sensor data."""
         try:
-            data = self.check_sensors()
-            if data is not None:
-                self.data = self._schema(data)
-                return
-        except OSError as error:
-            if retry < 1:
-                self.data = None
-                _LOGGER.error(error)
-                return
-        except (vol.Invalid, vol.MultipleInvalid):
-            pass  # Continue quietly if device returned malformed data
-        if retry > 0 and self._auth():
-            self._update(retry - 1)
-
-    def _auth(self, retry=3):
+            data = await self.device.async_request(self.check_sensors)
+        except BroadlinkException:
+            return
         try:
-            auth = self.api.auth()
-        except OSError:
-            auth = False
-        if not auth and retry > 0:
-            return self._auth(retry - 1)
-        return auth
+            data = self._schema(data)
+        except (vol.Invalid, vol.MultipleInvalid):
+            return
```

Comment:
I think it would be interesting to make a new request if the device returns malformed data. How about that?

Reply:
In general I think retrying should be done conservatively when it comes to wireless communication. Retrying might event make the problem worse. But I don't know the broadlink device platform.

Reply:
Done.

Reply:
I'm sorry, I didn't update the page and just saw the message now. Do you want me to revert?

Reply:
I don't know the reason for the retry functionality so I can't advice on this.

Reply:
The idea behind the retry is to request the data again if it arrives malformed. It is not common for this to happen and we will probably never reach the third attempt.

I think the best option for checking the integrity would be to implement a checksum in the library and raise an exception in case of error. I did some tests and I know this is possible. But I will leave it for the future, since I have already changed a lot here. This code will keep us running smoothly for now.